File: java/models/Day.java
package models;

import java.util.ArrayList;
import java.util.List;

public class Day {

  String timestamp;
  double averagePrice;
  List<Bitcoin> bitcoins;
  float label;
  int number_of_bitcoin;


  public Day(Bitcoin bitcoin) {
    this.timestamp = bitcoin.getTimestamp();
    bitcoins = new ArrayList<>();
    this.label = -1;
    this.number_of_bitcoin = 0;
  }

  public void addBitcoin(Bitcoin bitcoin) {
    this.bitcoins.add(bitcoin);
    this.number_of_bitcoin = this.bitcoins.size();
  }

  public void calculateAveragePrice() {
    double sum_of_prices = 0;
    for (Bitcoin bitcoin : this.bitcoins) {
      sum_of_prices += bitcoin.getWeightPrice();
    }
    averagePrice = sum_of_prices / bitcoins.size();
  }

  public String getTimestamp() {
    return timestamp;
  }

  public void setLabel(float label) {
    this.label = label;
  }

  public double getAveragePrice() {
    return averagePrice;
  }

  @Override
  public String toString() {
    if (this.label != -1)
      return timestamp + "," + averagePrice + "," + label + '\n';
    else
      return timestamp + "," + averagePrice + "," + '\n';
  }
}

File: cpp/models/Day.cpp
//
// Created by Christian C Lei on 2/19/21.
//
#include <iostream>
#include "Day.h"

Day::Day() {
    empty = true;
}

Day::Day(Bitcoin bitcoin) {
    empty = false;
    label = -1.0;
    timestamp = bitcoin.getTimestamp();
    number_of_bitcoin = 0;
}

const string &Day::getTimestamp() const {
    return timestamp;
}

void Day::add_bitcoin(Bitcoin bitcoin)
{
    bitcoins.push_back(bitcoin);
    number_of_bitcoin = bitcoins.size();
}

void Day::calculate_average_weighted_price()
{
    float sum_of_weighted_price = 0.0;
    float num_of_bitcoin_per_day = bitcoins.size();
    for (const Bitcoin& bitcoin : bitcoins)
    {
        sum_of_weighted_price += bitcoin.getWeightPrice();
    }

    average_price =  sum_of_weighted_price / num_of_bitcoin_per_day;
};

ostream& operator<<(std::ostream &strm, const Day &day) {
    strm.precision(5);
    if (day.label != -1.0)
        return strm << day.timestamp << "," << day.average_price << "," << day.label;
    else
        return strm << day.timestamp << "," << day.average_price << ",";
}

bool Day::isEmpty() const {
    return empty;
}

float Day::getAveragePrice() const {
    return average_price;
}

void Day::setLabel(float label) {
    Day::label = label;
}
